server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundSummaryServiceTests.java

```diff
@@ -0,0 +1,251 @@
+        .build();
+
+    ClusterSettings enabledClusterSettings = new ClusterSettings(enabledSummariesSettings, ClusterSettings.BUILT_IN_CLUSTER_SETTINGS);
+    ClusterSettings disabledDefaultEmptyClusterSettings = new ClusterSettings(
+        disabledDefaultEmptySettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+    ClusterSettings enabledButNegativeIntervalClusterSettings = new ClusterSettings(
+        enabledButNegativeIntervalSettings,
+        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS
+    );
+
+    // Construction parameters for the service.
+
+    DeterministicTaskQueue deterministicTaskQueue;
+    ThreadPool testThreadPool;
+
+    @Before
+    public void setUpThreadPool() {
+        deterministicTaskQueue = new DeterministicTaskQueue();
+        testThreadPool = deterministicTaskQueue.getThreadPool();
+    }
+
+    /**
+     * Test that the service is disabled and no logging occurs when
+     * {@link AllocationBalancingRoundSummaryService#ENABLE_BALANCER_ROUND_SUMMARIES_SETTING} defaults to false.
+     */
+    public void testServiceDisabledByDefault() {
+        var service = new AllocationBalancingRoundSummaryService(testThreadPool, disabledDefaultEmptyClusterSettings);
+
+        try (var mockLog = MockLog.capture(AllocationBalancingRoundSummaryService.class)) {
+            /**
+             * Add a summary and check it is not logged.
+             */
+
+            service.addBalancerRoundSummary(new BalancingRoundSummary(50));
+            service.verifyNumberOfSummaries(0); // when summaries are disabled, summaries are not retained when added.
+            mockLog.addExpectation(
+                new MockLog.UnseenEventExpectation(
+                    "Running balancer summary logging",
+                    AllocationBalancingRoundSummaryService.class.getName(),
+                    Level.INFO,
+                    "Balancing round summaries:*"
+                )
+            );
+
+            deterministicTaskQueue.runAllRunnableTasks();
+            mockLog.awaitAllExpectationsMatched();
+            service.verifyNumberOfSummaries(0);
+        }
+    }
+
+    public void testEnabledService() {
+        var service = new AllocationBalancingRoundSummaryService(testThreadPool, enabledClusterSettings);
+
+        try (var mockLog = MockLog.capture(AllocationBalancingRoundSummaryService.class)) {
+            /**
+             * Add a summary and check the service logs a report on it.
+             */
+
+            service.addBalancerRoundSummary(new BalancingRoundSummary(50));
+            service.verifyNumberOfSummaries(1);
+            mockLog.addExpectation(
+                new MockLog.SeenEventExpectation(
+                    "Running balancer summary logging",
+                    AllocationBalancingRoundSummaryService.class.getName(),
+                    Level.INFO,
+                    "Balancing round summaries:*"
+                )
```

Comment:
alternatively we could re-use the pattern from the seen assertion (via a constant), that way we'll be forced to keep the constant in sync with the message.

Reply:
Do you mean putting "Balancing round summaries:*" into a variable to deduplicate the string hardcoding? I've gone ahead with that interpretation.

Reply:
I was thinking of using the same constant for seen and unseen (i.e. if it stops working for seen it will fail the tests and it if works for seen it should also work for unseen). But wildcard is probably better.